integration/shareddata/benchmark.go

```diff
@@ -48,3 +58,37 @@ func generateMixedValues() func() bson.D {
 
 	return gen
 }
+
+// generateLargeDocument returns generator that generates single deterministic document
+// with 200 elements that contain simple strings, doubles and objects.
```

Comment:
Maybe not yet, I'm seeing `generateLargeDocument` and `generateLargeDocument` are very similar. It uses the same values to produce the documents, the subtle difference is the field key, `generateLargeDocument`  produces `{_id: "0", "v":<<val1>>}` and `generateLargeDocument` produces  `{_id: "0", "0":<<val1>>, "1":<<val2>>,...}`. When we add more generator we can think of merging those two functions and just have one with argument like field length and document length. But okay for now.

Reply:
Good point! I believe we could extract generator into separate interface and made them more general with some init functions like the one from your example!